### PR TITLE
[CBRD-22540] Add support for json_path wildcards

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-22540

Reimplement json_search, json_extract, json_contains_path to use JSON_TREE_FUNCTION.

### CODE REVIEW

File: src/xasl/access_json_table.cpp [hunk 1]
```diff
@@ -353,7 +323,7 @@ namespace cubxasl
 	    }
 	  else if (m_expand_type == json_table_expand_type::JSON_TABLE_OBJECT_EXPAND)
 	    {
-	      m_iterator = db_json_create_iterator (DB_JSON_TYPE::DB_JSON_OBJECT);
+	      assert (false);
```
Comment (ribram) [inline src/xasl/access_json_table.cpp]: remove all code related to JSON_TABLE_OBJECT_EXPAND. there is only expand or no expand.

File: src/compat/db_json.cpp [hunk 2]
```diff
@@ -3084,6 +3139,10 @@ db_json_sql_path_is_valid (std::string &sql_path, bool allow_wildcards)
 	      break;
 	    default:
 	      // unquoted object_keys
+	      // we normalize object_keys by quoting them - e.g. $."objectkey" will be representant for $.objectkey
+	      sql_path.insert (i++, "\"");
```
Comment (ribram) [inline src/compat/db_json.cpp]: Move `sql_path.insert (i++, "\"");` to db_json_path_is_token_valid_unquoted_object_key.

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/query/arithmetic.c (4 incorrect indentation(s) found)
```
  5536 	{
  5537 	  db_json_delete_doc (source_doc);
  5538 	  return error_code;
  5539 	}
```
#### File: src/query/string_opfunc.c (3016 incorrect indentation(s) found)
```
  1910   *dest_size = (int) (src_size + special_idx.size () + 2 /* quotes */  + 1 /* string terminator */ );
  3120 db_json_delete_doc (doc);
  3121 goto end;
  3122 const char *path;
  3123 db_json_delete_doc (doc);
  3124 goto end;
  3125 const char *path = db_get_string (arg[i]);
  3126 db_json_delete_doc (doc);
  3127 goto end;
  3128 paths.push_back (path);
  3129 }
  3130 
  3131 error_code = db_json_contains_path (doc, path, exists);
  3132 db_json_delete_doc (doc);
  3133 goto end;
  3134 
  3135 std::vector < std::string > paths;
  3136 if (wild_card_present || starting_paths.empty ())
  3137   std::vector < std::string > default_start_path (1, "$");
  3138 
```
... More than 20 lines are omitted.
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/query/arithmetic.c (4 incorrect indentation(s) found)
```
  5536 	{
  5537 	  db_json_delete_doc (source_doc);
  5538 	  return error_code;
  5539 	}
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by radudoros** [comment]
I've removed "in progress tag", please review, @ribram @eseokoh

**Pull request merged**